lidder.go

```diff
@@ -212,12 +224,74 @@ func main() {
 		}
 	}
 
-	if testFailed {
+	return success
+}
+
+func singleFileMode(configFilename, filename string) bool {
+	results := parseConfig(configFilename)
+
+	if !results.shouldCheck(filename) {
+		return true
+	}
+
+	results.adjustExpectedFilenames(filename)
+	err := results.matchAgainstFile(filename)
+	if err != nil {
+		oops(err)
+	}
+
+	success := true
+	for _, rule := range results.Rules {
+		shouldNotBeThere, shouldBeThere := rule.Mismatches()
+		if len(shouldNotBeThere) != 0 || len(shouldBeThere) != 0 {
+			success = false
+			if len(shouldNotBeThere) != 0 {
+				fmt.Printf("Lidded pattern '%s' found\n", rule.Pattern)
+			} else if len(shouldBeThere) != 0 { // mutually exclusive for a single file
+				fmt.Printf("Lidded pattern '%s' expected but not found\n", rule.Pattern)
+			}
+		}
+	}
+
+	return success
+}
+
+func parseConfig(filename string) *defs {
+	config, err := ioutil.ReadFile(filename)
+	if err != nil {
+		oops(err)
+	}
+
+	results, err := parse(config)
+	if err != nil {
+		oops(err)
+	}
+
+	return results
+}
+
+func main() {
+	var handler func(string, string) bool
+	var modePath string
+	if len(os.Args) == 2 {
+		handler = fullScanMode
+		modePath = "."
+	} else if len(os.Args) == 3 {
+		handler = singleFileMode
+		modePath = os.Args[2]
+	} else {
+		fmt.Println("usage: lidder config.yaml [file]")
+		fmt.Println("  -- If [file] is not specified, defaults to scanning all files from the current directory recursively")
```

Comment:
Can you make that usage a little nicer? My comment got lost
```
usage: lidder config.yaml [file]

  file              Optional. When specified, lidder only checks the specified
                     file.
```
(Line break to respect the convention that usage formats don't go over 80 chars.)